lighthouse-core/gather/gather-runner.js

```diff
@@ -187,17 +206,56 @@ class GatherRunner {
       const status = `Retrieving: ${gatherer.name}`;
       return chain.then(_ => {
         log.log('status', status);
-        return gatherer.afterPass(options, passData);
-      }).then(ret => {
+        const artifactPromise = Promise.resolve().then(_ => gatherer.afterPass(options, passData));
+        gathererResults[gatherer.name].push(artifactPromise);
+        return artifactPromise.catch(err => {
+          if (!err.recoverable) {
```

Comment:
seems a little weird to introduce the recoverable in this without any usage

will it be like
```
throw Object.assign(new Error('something'), {recoverable: true})
```

or will we have all throw statements be 3 lines?
```
const err = new Error('something');
err.recoverable = true;
throw err;
```